lib/matplotlib/tests/test_backend_svg.py

```diff
@@ -145,3 +144,79 @@ def test_svgnone_with_data_coordinates():
         buf = fd.read().decode()
 
     assert expected in buf
+
+
+def test_gid():
+    """Test that object gid appears in output svg."""
+    from mpl_toolkits.mplot3d import Axes3D  # noqa
+    from matplotlib.offsetbox import OffsetBox
+    from matplotlib.axis import Tick
+
+    fig = plt.figure()
+
+    ax1 = fig.add_subplot(131)
+    ax1.imshow([[1., 2.], [2., 3.]], aspect="auto")
+    ax1.scatter([1, 2, 3], [1, 2, 3], label="myscatter")
+    ax1.plot([2, 3, 1], label="myplot")
+    ax1.legend()
+    ax1a = ax1.twinx()
+    ax1a.bar([1, 2, 3], [1, 2, 3])
+
+    ax2 = fig.add_subplot(132, projection="polar")
+    ax2.plot([0, 1.5, 3], [1, 2, 3])
+
+    ax3 = fig.add_subplot(133, projection="3d")
+    ax3.plot([1, 2], [1, 2], [1, 2])
+
+    fig.canvas.draw()
+
+    counter = [0]
+
+    def assign_gid(obj):
+        gid = f"test123{obj.__class__.__name__}_{counter[0]}"
+        obj.set_gid(gid)
+        counter[0] += 1
+        return gid
+
+    gdic = {}
+
+    def walk(obj):
+        if obj.get_visible():
+            gdic[assign_gid(obj)] = obj
+        if hasattr(obj, "get_children"):
+            for child in obj.get_children():
+                walk(child)
+
+    walk(fig)
+
+    fd = BytesIO()
+    fig.savefig(fd, format='svg')
+    fd.seek(0)
+    buf = fd.read().decode()
+    fd.close()
+
+    def include(gid, obj):
```

Comment:
perhaps `has_gid()` or `gid_is_set()` (it's not totally clear what "include" means here)

Reply:
This is really literally, whether or not to include an object in the test. (Not all objects in the figure actually appear in the svg, so we need to exclude them). I've added a docstring to it.